### ISSUE TITLE
[Question] Codepush on a different computer, fresh install of project

### ISSUE THREAD

**Issue by LuisRizo** [Microsoft/react-native-code-push#1326]
My company released v1 of the app and is already available to the public. We are trying to release v2 and deprecate v1. The next step is to do a CodePush release on v1 to guide the users to download the new version v2. However, I did not develop v1 and I am not able to get it to build because of all the dependencies and some errors that are not worth fixing for the matter. 
I removed the dependencies and was able to run the app on an empty view, which is enough for me to redirect the user to download the new version.
My main question is, am I able to do a CodePush using a new environment with all the third party dependencies (including dependencies with native code) removed as if I'm pushing a brand new app, form a new machine. Is there other approach I should take?

[Posted on Stack Overflow as well](https://stackoverflow.com/questions/51068450/codepush-on-a-different-computer-fresh-install-of-project)

**Comment by ajwhite** [comment]
Yes. Since it's not the other way around (adding _new_ dependencies that require native modules), you're not accessing any _native features_ that don't already exist. You're just removing access to the already existing native features.

So given that, you should be in the clear.

**Comment by LuisRizo** [comment]
Just to summarize, in theory I can have an app as developed and with as many features as for example Instagram, with `code-push` installed, and at any point in time I can just edit the `index.js` file to just return do `AppRegistry.registerComponent('instagram', () => React.Fragment);`, delete all the imports in `index.js` and only leave `react` and `react-native` as dependencies in the `package.json` file and a `code-push` will work?

**Comment by LuisRizo** [comment]
Hi @alexandergoncharov,
Thank you for your reply!

My goal is to use a `code-push` to deprecate v1 of our app and guide our users to update to v2 since we'll also deprecate the back-end hooked up to v1. My dependencies are just enough to render a simple screen saying to the users that they should update. That includes `react`, `react-native` and, of course, `code-push`. I did an `npm install` so that should take care of the peer dependencies of those packages.

The reason behind changing native code is that the dependencies are old and some of them have been deprecated and cause bugs that are not worth fixing. I am **not** using **new** native code, I just uninstalled it to get the app to run on `ios` and `android` without compilation errors.

So if deleting, and stopping using native code will **not** cause my `code-push` to be rolled back automatically, I think I'm all set!

Please let me know if something isn't clear, or not correct

**Comment by alexandergoncharov** [comment]
Hi @LuisRizo ,
Sorry for long delay.

So, if I understand you correctly, you would like to remove some native code from your app. 
For this you should make new binary version with this changes and deliver it to your end-users. Before delivering you can also check that code-push work correct and try to update app on you test device. If I understand you correct you can sent some instructions to users with your app for getting this new binary version.  And then you can continue make updates with codepush.   

Please let me know if it was helpful or I understand your issue incorrect or you have any questions.

Thanks,
Alexander

**Comment by alexandergoncharov** [comment]
Hi @LuisRizo,
Thanks for this details!

Yeah, I got it. So, I think that it should works. It should not broke app If your new update will not use some installed plugin in your `v1.4.0` app.
But I think it will be better if you'll test installing this update on your local device to ensure that all this works correct in your case if it is possible.

I just tested it with test project and didn't find some issues.

Please let me know if it was helpful for you and then results of your work.

Thanks,
Alexander

**Comment by LuisRizo** [comment]
Hello @alexandergoncharov,

Thank you for your time and your response!

I appreciate you tested it locally! I'm really happy you found no issues. We're going to be doing this deployment soon so I will definitely let you know how it goes!

Thank you very much again for all.

Best,
Luis Rizo

**Comment by alexandergoncharov** [comment]
@LuisRizo I was glad to help.  You are always welcome :)

So, if question is answered I'm going to close this issue for now but please feel free to reopen it if you have any questions.

And I will be waiting your result. 

Thanks,
Alexander

**Comment by LuisRizo** [comment]
UPDATE: we tried doing the code-push update with that snippet code above, and it worked out great!

The live version of the app had rn-splash-screen though, and I did forget to put the code to hide it. A simple `npm install` of the library using the same version as app `v1.4.0` worked out beautifully as the native config for it was already in place.

One more thing that I noticed is that there is no way to actually rollback a release. Rollback just prevents future installations, I was hoping it would uninstall the last update on all the affected users.
